flake.nix

```diff
@@ -3,40 +3,62 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
-    flake-utils.url = "github:numtide/flake-utils";
     ixx = {
       # match version with npm package
       url = "github:NuschtOS/ixx/v0.0.6";
       inputs = {
         nixpkgs.follows = "nixpkgs";
-        flake-utils.follows = "flake-utils";
       };
     };
   };
 
-  outputs = { nixpkgs, flake-utils, ixx, ... }:
-    flake-utils.lib.eachDefaultSystem
-      (system:
+  outputs =
+    { nixpkgs, ixx, ... }:
+    let
+      inherit (nixpkgs) lib;
+      systems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+
+      forAllSystems = lib.genAttrs systems;
+      nixpkgsFor = nixpkgs.legacyPackages;
+    in
+    {
+      devShells = forAllSystems (
+        system:
         let
-          pkgs = (import nixpkgs) {
-            inherit system;
-          };
-          ixxPkgs = ixx.packages.${system};
+          pkgs = nixpkgsFor.${system};
```

Comment:
Constructing pkgs here for each output is slightly slowered when evaluating all things but that is normally not done.